### DISCUSSION TITLE
VS code not working from zsh but works from bash WSL

### DISCUSSION THREAD

**Discussion by raysaik** [ohmyzsh/ohmyzsh#12277] [Help] [answered]
I am using Ubuntu-22.04 as my WSL distro. For some reason only VSCode does not seems to work when i switch to zsh. THe error i am seeing is as below

```
Warning: unknown mime-type for "-a" -- using "application/octet-stream"
Warning: unknown mime-type for "Visual Studio Code" -- using "application/octet-stream"
Error: no such file "-a"
Error: no such file "Visual Studio Code"
```

However when i switch back to Ubuntu teeminal using "bash" command, it works fine
![image](https://github.com/ohmyzsh/ohmyzsh/assets/24606602/b912829b-7347-4c9f-985c-64d637b28b23)

**Comment by mcornella** [comment, marked as answer]
It looks to me like `code` thinks it's running under macOS and running `open -a "Visual Studio Code"`, which is used to open VS Code within macOS.

I don't see we are doing anything like that within Oh My Zsh, and searching on the official repo https://github.com/microsoft/vscode I don't see them doing anything like that either. I also didn't see it in the `code` command I have installed under Ubuntu 22.04 with version `1.92.0-insider` of VS Code.

It's possible that you have some alias or function for `code`, for which you could try `which -a code`, or in any case inspect the script that it is running with `cat "${commands[code]}"`.

If it turns out to be a VS Code issue, then gather the information necessary and submit an issue in the upstream repo: https://github.com/microsoft/vscode/issues/new/choose.
